SUS-4512 | Dockerfile for MediaWiki PHP app container

macbre opened the pull request:
https://wikia-inc.atlassian.net/browse/SUS-4512

This is an initial version of `Dockerfile` to run Wikia app and make all tests pass. Once we complete this step, we can create a base PHP image out of this one and then build prod and dev images.

### Unit tests pass

```
# sh php-all
Time: 1.02 minutes, Memory: 164.00MB

OK, but incomplete, skipped, or risky tests!
Tests: 4344, Assertions: 6740, Skipped: 19.

--

Time: 51.08 seconds, Memory: 108.00MB

There was 1 failure:

1) UploadVerifyFile::testUploadVerifyFile with data set #1 ('image/gif', false, 'foobar')
Failed asserting that true matches expected false.

/usr/wikia/slot1/current/src/includes/wikia/tests/UploadVerifyFileTest.php:38

FAILURES!
Tests: 201, Assertions: 769, Failures: 1.
```

There's just a single failing integration test (it checks the code that calls `identify` binary).

macbre committed:
SUS-4512 | initial commit of Dockerfile for MediaWiki PHP app container

macbre committed:
Dockerfile: install mysqli PHP extension

macbre committed:
run-test: increase memory limit to 256MB explicitily here

macbre committed:
Dockerfile: use Debian-powered PHP base image + install make

macbre committed:
Compile and set up sassphp

macbre committed:
Compile and set up mustache PHP module

macbre committed:
Compile and set up uopz extension

macbre committed:
Compile and set up wikidiff2 extension

macbre committed:
Dockerfile: run with "localhost" as a hostname

macbre committed:
Set up bz2 extension

macbre committed:
Merge branch 'dev' into SUS-4512

macbre committed:
Add Zend OPcache module

php -v
PHP 7.0.28 (cli) (built: Mar 14 2018 15:45:16) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.28, Copyright (c) 1999-2017, by Zend Technologies

macbre committed:
Dockerfile: expose a volume with localisation cache files

macbre committed:
Docker.md: installation instructions

Commit:
Merge commit 'da75702aa16b5dd7d9be9c4e7bb97addb76d7a59' into SUS-4512

macbre committed:
Docker.md: simplify devbox install steps

macbre committed:
Unit tests in a container: mock variable in AdEngine2ContextServiceTest

pwojcik86 reviewed (dismissed):
We also need a container that has a fixed version of the code inside, which we could deploy on a k8s cluster. The one you've created is suited for development only.

macbre committed:
Dockerfile: add xdebug extension

macbre committed:
Dockerfile: set locales

mszabo-wikia commented on Dockerfile:
```diff
@@ -0,0 +1,85 @@
+
+# uopz extension / @see https://pecl.php.net/package/uopz
+RUN pecl install uopz-5.0.2 \
+    && docker-php-ext-enable uopz
```
should we install uopz in production?

macbre committed:
WikiFactoryLoader: it makes no sense to log debug messages via error_log() on devboxes

macbre committed:
Merge branch 'dev' into SUS-4512

macbre committed:
Dockefile: mark uopz and xdebug extension as dev only

macbre committed:
WikiaDatabaseTest: disable tasks queue when integration tests are run

macbre committed:
Docker: small README update

macbre committed:
Dockerfile: add yaml extension

macbre committed:
Docker.md: remove realpath

macbre committed:
Dockerfile: use Wikia forks of git repos to build a base image

macbre committed:
ImageServingCroppingTest: remove the part that fetched the file and checks its dimensions

This part is covered by Vignette tests - https://github.com/Wikia/vignette/blob/1890639494dba37dafed7b9df2bc4cfd30f5f34b/test/vignette/http/integration_test.clj

macbre committed:
Docker.md: document how to push images to artifactory

macbre committed:
Link to docker.com documentation

macbre committed:
Change Docker image names and tags
